src/engine/TransitivePath.cpp

```diff
@@ -381,11 +387,13 @@ void TransitivePath::computeTransitivePathLeftBound(
   size_t last_result_begin = 0;
   size_t last_result_end = 0;
   for (size_t i = 0; i < left.size(); i++) {
+    checkTimeoutAfterNCalls();
     if (left[i][leftSideCol] == last_elem) {
       // We can repeat the last output
       size_t num_new = last_result_end - last_result_begin;
       size_t res_row = res.size();
       res.resize(res.size() + num_new);
+      checkTimeoutAfterNCalls(num_new);
```

Comment:
Why not num_new * res_width ?

I presume that res_width is never very large. But what about num_new?

Reply:
Num new can in theory be very large (but this means, that the transitive path is easy to compute and mostly only copies.